bitnami/mongodb/templates/standalone/dep-sts.yaml

```diff
@@ -180,18 +180,25 @@ spec:
           {{- else if .Values.command }}
           command: {{- include "common.tplvalues.render" (dict "value" .Values.command "context" $) | nindent 12 }}
           {{- end }}
+          command:
+            - /scripts/setup.sh          
           {{- if .Values.diagnosticMode.enabled }}
           args: {{- include "common.tplvalues.render" (dict "value" .Values.diagnosticMode.args "context" $) | nindent 12 }}
           {{- else if .Values.args }}
           args: {{- include "common.tplvalues.render" (dict "value" .Values.args "context" $) | nindent 12 }}
           {{- end }}
           env:
             - name: BITNAMI_DEBUG
-              value: {{ ternary "true" "false" (or .Values.image.debug .Values.diagnosticMode.enabled) | quote }}
+              value: {{ ternary "true" "false" .Values.image.debug | quote }}
+            {{- if (and .Values.auth.enabled (not .Values.auth.vaultCredentials)) }}
             {{- if .Values.auth.username }}
             - name: MONGODB_USERNAME
               value: {{ .Values.auth.username | quote }}
             {{- end }}
+            {{- if .Values.auth.userRole }}
+            - name: MONGODB_USER_ROLE
+              value: {{ .Values.auth.userRole | quote }}
+            {{- end }}
```

Comment:
I don't think this env. variable is being taken into account.

Reply:
This is to add another fix. When the custom user created in a container can have a custom role.  As of now, the mongo container is creating a custom user with only read-write access to one database. With this parameter, we can assign a custom role to the Mongo custom user. Please refer the below PR.
https://github.com/bitnami/bitnami-docker-mongodb/pull/281